apps/sms/test/marionette/lib/views/new-message/view.js

```diff
@@ -77,11 +102,73 @@ NewMessageView.prototype = {
   },
 
   typeMessage: function(message) {
-    this.accessors.messageInput.sendKeys(message);
+    this.composerAccessors.messageInput.sendKeys(message);
+  },
+
+  addAttachment: function() {
+    return this.composerAccessors.addAttachment();
+  },
+
+  typeSubject: function(subject) {
+    this.composerAccessors.subjectInput.sendKeys(subject);
+  },
+
+  showOptions: function() {
+    this.accessors.optionsButton.tap();
+  },
+
+  showSubject: function() {
+    this.showOptions();
+    this.menuAccessors.selectAppMenuOption('Add subject');
+    this.client.helper.waitForElement(this.composerAccessors.subjectInput);
+  },
+
+  hideSubject: function() {
+    this.showOptions();
+    this.menuAccessors.selectAppMenuOption('Remove subject');
+    this.client.helper.waitForElementToDisappear(
+      this.composerAccessors.subjectInput
+    );
+  },
+
+  send: function() {
+    this.composerAccessors.send();
+
+    var ConversationView = require('../conversation/view');
+    var conversationView = new ConversationView(this.client);
+    conversationView.accessors.waitToAppear();
+
+    return conversationView;
+  },
+
+  backToInbox: function() {
+    this.accessors.header.scriptWith(function(header) {
+      var event = document.createEvent('HTMLEvents');
```

Comment:
Well, I don't remember, likely copy paste from our unit tests [1]. But after bug https://bugzilla.mozilla.org/show_bug.cgi?id=1206093, it won't really matter, depending on who lands first I'll either start using ```client.switchToShadowRoot``` in this patch or ask contributor to update code here as well :)

[1] https://github.com/mozilla-b2g/gaia/blob/e42f7e9937b62f84d2baef547f5bd08e1a6ac44b/apps/sms/views/conversation/test/unit/conversation_test.js#L2543